File: Cpp_vs_Python/TeuflischeFolge.cpp
// =====================================================================================
// TeuflischeFolge.cpp
// =====================================================================================

#include <iostream>
#include <print>
#include <string>

static void berechne_teuflische_folge(int start);

void example_teuflische_folge()
{
    std::println("Teuflische Zahlenfolge");
    std::println("======================");
    std::println();
    
    std::print("Startwert: ");
    std::string input;
    std::cin >> input;
    int number = std::stoi(input);

    berechne_teuflische_folge(number);
}

void berechne_teuflische_folge(int start)
{
    auto number{ start };   // start number
    auto n{ 1 };            // counter for length of sequence

    while (number != 1)
    {
        if (number % 2 == 0)
        {
            number = number / 2;
        }
        else
        {
            number = 3 * number + 1;
        }

        std::println("Schritt {:3}: Zahl = {}", n, number);
        n++;
    }
}

// =====================================================================================
// End-of-File
// =====================================================================================

File: Cpp_vs_Python/TeuflischeFolge/TeuflischeFolge.py
# =====================================================================================
# TeuflischeFolge.py
# =====================================================================================

def example_teuflische_folge(start_number=7):
    print("Teuflische Zahlenfolge\n")
    print("======================\n")

    number = start_number   # Startzahl
    n = 1                   # Zaehler fuer die Laenge der Folge

    print(f"Startzahl: {number}")
    print()

    while number != 1:
        if number % 2 == 0:
            number //= 2                # Ganzzahl-Division
        else:
            number = 3 * number + 1

        print(f"Schritt {n:3}: Zahl = {number}")
        n += 1

if __name__ == "__main__":
    example_teuflische_folge(7)

if __name__ == "__main__":
    # Benutzereingabe in eine int-Variable umwandeln
    value = int(input("Startzahl: "))
    example_teuflische_folge(value)

# =====================================================================================
# End-of-File
# =====================================================================================
